From 5e65513c3be5ee5d464017bcb5c9d6222228b7f4 Mon Sep 17 00:00:00 2001
From: Prasanth Babu Mantena <p-mantena@ti.com>
Date: Tue, 25 Apr 2023 17:15:04 -0500
Subject: [PATCH 407/508] vxe-vxd: decoder: Fix input buffer size

Optimize decoder Input buffer size. The Input buffer was
allocated with size equal to raw frame size of YUV422 which is not supported
by the driver. The frame size is mainly dependent on the
resolution (width and height) and also the bit depth (8/10/12 etc).
As we only support NV12 format, width * height * 1.5 should be sufficient
for holding an encoded frame.

Signed-off-by: Prasanth Babu Mantena <p-mantena@ti.com>
Signed-off-by: Darren Etheridge <detheridge@ti.com>
---
 drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c b/drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c
index b622134d6377..9b9645416fbc 100644
--- a/drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c
+++ b/drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c
@@ -1379,11 +1379,10 @@ static int vxd_dec_try_fmt(struct file *file, void *priv, struct v4l2_format *f)
 			return -EINVAL;
 		}
 		/*
-		 * Allocation for worst case input frame size:
-		 * I frame with full YUV size (YUV422)
+		 * Allocation for NV12 input frame size:
 		 */
-		plane_fmt[0].sizeimage = ALIGN(pix_mp->width, HW_ALIGN) *
-			ALIGN(pix_mp->height, HW_ALIGN) * 2;
+		plane_fmt[0].sizeimage = (ALIGN(pix_mp->width, HW_ALIGN) *
+			ALIGN(pix_mp->height, HW_ALIGN) * 3)/2;
 	} else {
 		fmt = find_format(f, IMG_DEC_FMT_TYPE_CAPTURE);
 		if (!fmt) {
-- 
2.41.0

